src/pipx/interpreter.py

```diff
@@ -56,7 +57,23 @@ def _find_default_windows_python() -> str:
     return python  # This executable seems to work.
 
 
-if WINDOWS:
-    DEFAULT_PYTHON = _find_default_windows_python()
+def _get_sys_executable() -> str:
+    if WINDOWS:
+        return _find_default_windows_python()
+    else:
+        return sys.executable
+
+
+def _get_absolute_python_interpreter(env_python: str) -> str:
+    which_python = shutil.which(env_python)
+    if not which_python:
+        raise PipxError(f"Default python interpreter '{env_python}' is invalid")
```

Comment:
`{env_python!r}` or `{repr(env_python)}` would handle the quoting edge cases better.

Reply:
Hi, would you mind giving an example of a quoting edge case that `repr` would fix? I thought that f-strings already handled escaping, and since `env_python` is always a string, `repr` seems redundant to me. Happy to add either way, but I just want to understand if I'm missing something.

Reply:
The main difference is when the path contains a quote character. `repr()` would automatically choose the best quoting scheme, and perform better than the current naive method of simply adding quote characters around the path.

Alternatively you can just not quote the path, like you did for the “python interpreter is” message below. That would cause some confusion when there’s space in the path, but there’s no perfect solution here, unfortunately.

Reply:
Ah, okay. Thank you for explaining!